### ISSUE TITLE
Question Set not part of Course Presentation

### ISSUE THREAD

**Issue by supriyarajgopal-spi** [h5p/h5p-course-presentation#22]
Course Presentation is indeed very powerful but it lacks inclusion of Question Set content type.
I have modified the Course Presentation 1.10 & added a minor condition in H5P.QuestionSet1.8\js\questionset.js to make this feature available.

Modified files:
- H5P.QuestionSet-1.8\js\questionset.js
- H5P.CoursePresentation-1.10\semantics.json
-H5PEditor.CoursePresentation-1.10\library.json

Please find attached the above 3 libraries.
[QuestionSet_CoursePresentation.zip](https://github.com/h5p/h5p-course-presentation/files/388313/QuestionSet_CoursePresentation.zip)

**Comment by thomasmars** [comment]
This is interesting indeed. I have created an issue for reviewing if Question Set fits inside Course Presentation and code feedback. Hopefully we can have a look at this in the next sprint.
Thanks

**Comment by supriyarajgopal-spi** [comment]
Thanks for the quick response @thomasmars . Any idea when would the next sprint end so we can have this in the release if possible?

**Comment by thomasmars** [comment]
We are in the final stages of our current release. We try to release monthly, so probably a month from here, but this may vary.

**Comment by thomasmars** [comment]
The merge to master may happen a lot faster though

**Comment by supriyarajgopal-spi** [comment]
Great...I hope to see this included in the next release, thank you.

**Comment by falcon-git** [comment]
This might take a bit longer. We're considering to make a mini questionset aimed at this. The current questionset might be too advanced already and it will also be harder to continue the development of it if it is included in Course Presentation. Among other things CoursePresentation has a fixed height and if we in a future update makes question set taller we might make existing Course Presentations with question sets in them look ugly with extra scroll bars.

**Comment by supriyarajgopal-spi** [comment]
I agree @falcon-git, I already notice the scroll bars in the Course Presentation in the hack that I have done to include Question Set in CP. But the themer in our team may work on it to fix the UI issues.

Hoping to see this functionality included in CP sometime in the near future.

**Comment by supriyarajgopal-spi** [comment]
@falcon-git , I need your assistance on removing those scroll bars when a question set is used inside a Course Presentation. Is there a way I can make Course Presentation's height expand according to its content inside?

**Comment by thomasmars** [comment]
I'm not sure why you're experiencing scroll bars in question set and drag&drop, is it because they are too big for the Course Presentation or is it because of a styling issue ? Either way the solution is not to increase Course Presentation's height, because all of the scaling logic of CP is built around it having a set ratio. I would approach the issue through either scaling the content inside QS and DnD or be content with scroll bars.

**Comment by supriyarajgopal-spi** [comment]
Hello @thomasmars ,

It is not a styling issue for now. I have a Drag&Drop inside a CP with almost 10 draggable elements. This adds a scrollbar to it. Scaling the content inside D&D makes the font size too small and not legible.
Do you mean these are the only two alternatives?Is there no way to increase the height of CP??

**Comment by thomasmars** [comment]
There might be some other alternatives, but increasing the height of CP should not be done and is not possible to my knowledge.

**Comment by supriyarajgopal-spi** [comment]
Sigh.. :( Ok @thomasmars , I will see what best I can do with what I have. But if there is any progress/solution or if you consider this as a recurring scenario, please do update with a solution.

Thanks for the guidance, :+1:

**Comment by supriyarajgopal-spi** [comment]
Me again.
I hope this is the right place to raise this issue.
I observed that when Question Set inside Course Presentation, the 'completed' verb never occurs. The verb remains 'answered' even on clicking Finish button.
This doesn't happen when Question Set is used standalone.
If I comment out the following snippet from line 27 in navigation-line.js, all works well:
`
else if (shortVerb === 'completed') {
              event.setVerb('answered');
            }
`
I see a comment 'An activity within this activity is not allowed to send completed events' but I do not understand its significance. Can I comment out it without any side effects?Could anybody explain why is it force set to 'answered'?

Regards,
Supriya Rajgopal

**Comment by supriyarajgopal-spi** [comment]
I figured out the side effect:
The setFinished() function in h5p.js is not called at all if I comment out the above mentioned line.
To fix this, I am pasting the following line in my custom .js file:
`H5P.setFinished(contentId, event.data.score, event.data.maxScore);`

I hope this is correct. But this causes another JS error:
'H5P.opened[contentId]' is undefined in setFinished(). 

Can anybody throw some light on how to fix this one JS error please?

**Comment by fnoks** [comment]
I guess completed should be sent once at the end of the Course Presentation (when the summary slide is shown)?

**Comment by supriyarajgopal-spi** [comment]
Hello @fnoks ,
Thanks for responding. It makes sense to consider it completed when the Summary Slide is shown. 
I am not sure if it works fine when I choose to Hide a summary slide.

**Comment by fnoks** [comment]
If summary is hidden, it probably should be sent on the last slide. But I am not sure the current imlpementation does this. Probably not, since knowing when the end user is finished is not possible to know.

**Comment by supriyarajgopal-spi** [comment]
Thanks for the clarification @fnoks . I'll see what best can be done without hacking the core or wait for a release that fixes this if it is valid.

**Comment by otacke** [comment]
Hi! I was made a community maintainer for H5P Group's repositories a few weeks ago, and I am still in the process trying to tackle all the open issues and pull requests.

I am terribly sorry that this issue went stale.

I don't know what the opinion about QuestionSet in Course Presentation is nowadays. So, I'll move this to "verified by community maintainer" (see https://github.com/orgs/h5p/projects/6/), meaning it is supposed to be prioritized by H5P Group. Please note, however, that I don't know when they are going to handle this. My powers end here.
